### PR TITLE
GH-433: Exception in JSXBackendHelper in case project "react" exists in workspace

### PR DESCRIPTION
**Pull request by mor-n4**
See #433.

### CODE REVIEW

**Comment by mor-n4** [comment]
builds are green

File: plugins/org.eclipse.n4js.transpiler/src/org/eclipse/n4js/n4jsx/transpiler/utils/JSXBackendHelper.java [hunk 1]
```diff
@@ -168,74 +139,33 @@ public String jsxBackendModuleQualifiedName(Resource resource) {
...
-	 */
-	private static URI stubMerger(URI first, URI second) {
-		if (first.lastSegment().compareToIgnoreCase(second.lastSegment()) > 0)
-			return first;
-		return second;
+	private void addReactToBackendsCache(Resource resource) {
+		TModule reactModule = reactHelper.lookUpReactTModule(resource);
+		if (reactModule != null) {
+			URI reactModuleURI = reactModule.eResource().getURI();
+			jsxBackends.put(ReactHelper.REACT_PROJECT_ID, reactModuleURI);
```
Comment (yoosiba) [inline plugins/org.eclipse.n4js.transpiler/src/org/eclipse/n4js/n4jsx/transpiler/utils/JSXBackendHelper.java]: If we delegate to react helper, which is caching lookup result, do we need to cache here?

Comment (mor-n4) [reply plugins/org.eclipse.n4js.transpiler/src/org/eclipse/n4js/n4jsx/transpiler/utils/JSXBackendHelper.java]: The caching in JSXBackendHelper is problematic, anyway, and probably has to be removed; but this has to be investigated further as part of #358.

But I'll add a comment there to make sure it won't be forgotten!

**Comment by mor-n4** [comment]
thanks, merging ...

**Comment by mor-n4** [comment]
see https://github.com/eclipse/n4js/issues/358#issuecomment-354984772

**Pull request merged**